### PR TITLE
feat(react-grid): implement table column resizing

### LINKED ISSUE #192: Provide the capability to resize columns
Provide the capability to resize columns (similar to [allowColumnResizing](https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#allowColumnResizing)).

### PR DESCRIPTION
**Pull request by kvet**
fixes #192

### CODE REVIEW

File: packages/dx-grid-core/src/plugins/table-column-resizing/reducers.test.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      };
+
+      expect(changeDraftTableColumnWidths(state, { shifts: { b: -25 } }))
+        .toEqual({
+          columnWidths: { a: 40, b: 60 },
+          draftColumnWidths: { b: 40 },
+        });
+    });
+
+    it('should stick size to the min', () => {
```
Comment (gsobolev) [inline packages/dx-grid-core/src/plugins/table-column-resizing/reducers.test.js]: The previous assertion looks identical to this one.

File: packages/dx-grid-core/src/plugins/table-column-resizing/reducers.test.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      };
+
+      expect(changeDraftTableColumnWidths(state, { shifts: { b: -25 } }))
+        .toEqual({
+          columnWidths: { a: 40, b: 60 },
+          draftColumnWidths: { b: 40 },
+        });
+    });
+
+    it('should reset size whel null passed', () => {
```
Comment (gsobolev) [inline packages/dx-grid-core/src/plugins/table-column-resizing/reducers.test.js]: *when?

File: packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.jsx [hunk 3]
```diff
@@ -75,9 +102,71 @@ export class TableHeaderCell extends React.PureComponent {
       )
     );
 
+    const sesizingControlLineBody = resizing && (
```
Comment (gsobolev) [inline packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.jsx]: *resizingControlLineBody?

File: packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.jsx [hunk 4]
```diff
@@ -75,9 +102,71 @@ export class TableHeaderCell extends React.PureComponent {
...
+              backgroundColor: '#ddd',
+              height: '50%',
+              width: '1px',
+              left: '7px',
+              top: '25%',
+            }}
+          >
+            {sesizingControlLineBody}
+          </div>
+        </div>
```
Comment (gsobolev) [inline packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.jsx]: Can we split this somehow? Too many lines...

File: packages/dx-react-grid-material-ui/src/templates/table-header-cell.jsx [hunk 5]
```diff
@@ -86,6 +88,46 @@ const styles = theme => ({
     textAlign: 'left',
     marginRight: (theme.spacing.unit * 2) - 2,
   },
+  resizeHandle: {
+    position: 'absolute',
+    userSelect: 'none',
+    MozUserSelect: 'none',
+    WebkitUserSelect: 'none',
+    width: '16px',
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid-material-ui/src/templates/table-header-cell.jsx]: Use the `theme.spacing.unit` here.

File: packages/dx-react-grid-material-ui/src/templates/table-header-cell.jsx [hunk 6]
```diff
@@ -172,17 +241,33 @@ class TableHeaderCellBase extends React.PureComponent {
...
+      >
+        <div
+          className={classNames({
+            [classes.resizeHandle]: true,
+            [classes.resizeHandleActive]: resizing,
+          })}
+        >
+          <div
+            className={classes.resizeHandleLine}
+            style={{ left: '5px' }}
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid-material-ui/src/templates/table-header-cell.jsx]: Can replace with css class?

File: packages/dx-react-grid/src/plugins/table-column-resizing.test.jsx [hunk 7]
```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { setupConsole } from '@devexpress/dx-testing';
+import { PluginHost } from '@devexpress/dx-react-core';
+import {
+  tableColumnsWithWidths,
+  changeTableColumnWidths,
+  changeDraftTableColumnWidths,
+  cancelTableColumnWidth,
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid/src/plugins/table-column-resizing.test.jsx]: Can't find `cancelTableColumnWidth` in dx-grid-core

File: packages/dx-react-grid-material-ui/src/templates/table-header-cell/resizing-control.jsx [hunk 8]
```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Draggable } from '@devexpress/dx-react-core';
+import { withStyles } from 'material-ui/styles';
+
+const styles = theme => ({
+  // cell: {},
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid-material-ui/src/templates/table-header-cell/resizing-control.jsx]: Unexpected comment)

File: packages/dx-react-grid/src/plugins/table-column-resizing.jsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+      const { onColumnWidthsChange } = this.props;
+      if (onColumnWidthsChange) {
+        onColumnWidthsChange(nextState.columnWidths);
+      }
+    };
+
+    this.changeDraftTableColumnWidthsAction = (payload) => {
+      this.setState(changeDraftTableColumnWidths(this.getState(), payload));
+    };
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid/src/plugins/table-column-resizing.jsx]: Unnecessary curly braces

**Comment by wpascucci** [comment]
Hi guys is there any idea when this feature will be released ?

**Comment by kvet** [comment]
Hi @wpascucci,

We have plans to release this feature next week.

**Pull request merged**